winit-core/src/keyboard.rs

```diff
@@ -1732,6 +1733,29 @@ impl ModifiersState {
 }
 
 /// The logical state of the particular modifiers key.
+///
+/// NOTE: while the modifier can only be in a binary active/inactive state, it might be helpful to
+/// note the context re. how its state changes by physical key events.
+///
+/// `↓` / `↑` denote physical press/release[^1]:
+///
+///  | Type | Activated | Deactivated | Comment |
+///  | -----| :----: | :------: | ------- |
+///  |__Regular__ | `↓` | `↑`  | Active while being held |
+///  |__Sticky__ | `↓` | `↓` unless lock is enabled<br>`↓`/`↑`[^2] __non__-sticky key | Temporarily "stuck"; other `Sticky` keys have no effect|
+///  |__Sticky Locked__ | `↓` <br>if `Sticky` | `↓` | Similar to `Toggle`, but deactivating `↓` turns on `Regular` effect |
+///  |__Toggle__ | `↓` | `↓` | `↑` from the activating `↓` has no effect|
```

Comment:
Nit: Formatted to be easier to read.

```suggestion
/// | Type              | Activated           | Deactivated | Comment |
/// | ----------------- | :-----------------: | :---------: | ------- |
/// | __Regular__       | `↓`                 | `↑`         | Active while being held |
/// | __Sticky__        | `↓`                 | `↓` unless lock is enabled<br>`↓`/`↑`[^2] __non__-sticky key | Temporarily "stuck"; other `Sticky` keys have no effect |
/// | __Sticky Locked__ | `↓` <br>if `Sticky` | `↓`         | Similar to `Toggle`, but deactivating `↓` turns on `Regular` effect |
/// | __Toggle__        | `↓`                 | `↓`         | `↑` from the activating `↓` has no effect |
```